commands/user.go

```diff
@@ -104,30 +114,47 @@ func init() {
 	RootCmd.AddCommand(UserCmd)
 }
 
-func deactivateUsers(c client.Client, userArgs []string) {
+func userActivateCmdF(c client.Client, command *cobra.Command, args []string) error {
+	if len(args) < 1 {
+		return errors.New("Expected at least one argument. See help text for details.")
+	}
+
+	changeUsersActiveStatus(c, args, true)
+
+	return nil
+}
+
+func changeUsersActiveStatus(c client.Client, userArgs []string, active bool) {
 	users := getUsersFromUserArgs(c, userArgs)
 	for i, user := range users {
-		if user == nil {
-			printer.PrintError("Unable to find user '" + userArgs[i] + "'")
-			continue
-		}
+		err := changeUserActiveStatus(c, user, userArgs[i], active)
 
-		if user.IsSSOUser() {
-			printer.Print("You must also deactivate user " + userArgs[i] + " in the SSO provider or they will be reactivated on next login or sync.")
+		if err != nil {
+			printer.PrintError(err.Error())
 		}
+	}
+}
 
-		if _, response := c.DeleteUser(user.Id); response.Error != nil {
-			printer.PrintError("Unable to deactivate user " + userArgs[i] + ". Error: " + response.Error.Error())
-		}
+func changeUserActiveStatus(c client.Client, user *model.User, userArg string, activate bool) error {
+	if user == nil {
+		return fmt.Errorf("Can't find user '%v'", userArg)
+	}
```

Comment:
I'd prefer this check to be made right after getting the user, and for the `changeUserActiveStatus` function to trust the arguments so it's simpler and the errors are handled right where they occur. [3/5](https://docs.mattermost.com/process/training.html#out-of-5)

Reply:
If this is the only workflow that is expected (changeUsersActiveStatus->changeUserActiveStatus) this work. If at any moment we decide to expose or use in other place the "changeUserActiveStatus" function, it may result in a panic. That is why I generally prefer to check things just before using them.

Nevertheless, if we want to enforce that changeUserActiveStatus to trust the arguments and expect a panic if someone or something is using it wrong, we can make the change.

Reply:
For me is the other way around, if we retrieve something but don't check that we got a value, that could cause a panic. Now we are doing `changeUsersActiveStatus -> changeUserActiveStatus`, but if we decide to do something else with those users before (or even after) going through the function that checks the `nil`s, we would be operating with usafe values.

Checking something as part of getting it is safe, as it doesn't matter where you use it, you know that it has already been checked, hence the rest of the functions can trust that value (or check it for double safety if the functions are exposed as part of an API).